### PR TITLE
Update providesModuleNodeModules option to use new format.

### PR DESCRIPTION
**Pull request by skevy**
Updates providesModuleNodeModules option to use new format from https://github.com/facebook/node-haste/pull/33.

Supercedes https://github.com/facebook/react-native/pull/5985.

Ok to merge when node-haste is updated.

/cc @bestander @cpojer @mkonicek

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,11 +24,10 @@
...
       "providesModuleNodeModules": [
-        "fbjs",
-        "react",
-        "react-native",
-        "parse",
-        "react-transform-hmr"
+        { name: "fbjs" },
+        { name: "react" },
+        { name: "react-native" },
+        { name: "parse" }
```
Comment (mkonicek) [inline package.json]: Should name be in quotes as it's a JSON file?

Comment (skevy) [inline package.json]: Of course it should! :)

**Comment by davidaurelio** [comment]
@facebook-github-bot shipit

**Comment by davidaurelio** [comment]
I will hold this until Monday, though

**Comment by mkonicek** [comment]
Closing this one as it's superseded by #6346.

**Pull request closed without merging**